### PR TITLE
Support supplementaryViews created from storyboard

### PR DESCRIPTION
**Pull request by zhubofei**
Miss this in the previous PR #92. Will add unit test soon.

### CODE REVIEW

**Comment by rnystrom** [comment]
@zhubofei nice catch! Before we land this, can we add some unit tests to make sure they are working? Looks like you added views to the test storyboard, just need like one test and we're g2g.

**Comment by zhubofei** [comment]
@rnystrom I'm working on that right now and should finish by the end of the day. I do notice one thing, though. In the source code, all the *ItemControllers have been renamed to *SectionController. But in unit tests, they stay with *Item*. Do you keep their name on purpose?

**Comment by rnystrom** [comment]
@zhubofei ha, leftover from naming refactor. Surprised those never showed up! Yes we should move them in a separate PR. #126

**Comment by zhubofei** [comment]
A simple test with a lot of pain. Finally. 😂

**Comment by zhubofei** [comment]
@rnystrom Should I test that annoying 0.07% in StackSectionController? 😂

File: Source/IGListAdapter.m [hunk 1]
```diff
@@ -778,6 +778,21 @@ - (__kindof UICollectionReusableView *)dequeueReusableSupplementaryViewOfKind:(N
...
+                                                                             withIdentifier:(NSString *)identifier
+                                                                       forSectionController:(IGListSectionController <IGListSectionType> *)sectionController
+                                                                                    atIndex:(NSInteger)index {
+    IGAssertMainThread();
+    IGParameterAssert(elementKind.length > 0);
+    IGParameterAssert(identifier.length > 0);
+    IGParameterAssert(sectionController != nil);
+    IGParameterAssert(index >= 0);
+    UICollectionView *collectionView = self.collectionView;
+    IGAssert(collectionView != nil, @"Reloading adapter without a collection view.");
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: nit: We just updated our asserts in #122, can we make this assert read something like this?

https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.m#L720

**Comment by rnystrom** [comment]
@zhubofei 😂  no use, don't worry about it.

Small nit then we're g2g on this!

**Comment by rnystrom** [comment]
@zhubofei it looks like `IGListKit.xcodeproj/project.pbxproj` got out of sync. Mind rebasing and fixing that up?

**Comment by zhubofei** [comment]
@rnystrom Just did. I think we're g2g.

**Pull request closed without merging**